src/Support/SupportService.php

```diff
@@ -0,0 +1,187 @@
+	 *
+	 * @return array Support data.
+	 */
+	public function get_data( $args = [] ) {
+
+		$support = new SupportData( $args );
+
+		return $support->get_data();
+	}
+
+	/**
+	 * Sends support data to endpoint.
+	 *
+	 * ## OPTIONS
+	 *
+	 * [--is-synthetic]
+	 * : Whether or not it is synthetic data.
+	 * ---
+	 * default: false
+	 * options:
+	 *   - true
+	 *   - false
+	 *
+	 * [--print]
+	 * : To print support data.
+	 * ---
+	 * default: json-pretty
+	 * options:
+	 *   - json
+	 *   - json-pretty
+	 *
+	 * [--endpoint=<string>]
+	 * : Support endpoint. Where support data will send.
+	 *
+	 * [--urls=<urls>]
+	 * : List of URL for which support data need to send. Use comma separator for multiple URLs.
+	 *
+	 * [--post_ids=<post_ids>]
+	 * : List of Post for which support data need to send. Use comma separator for multiple post ids.
+	 *
+	 * [--term_ids=<term_ids>]
+	 * : List of term for which support data need to send. Use comma separator for multiple term ids.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     wp amp send-diagnostic
+	 *
+	 * @subcommand send-diagnostic
+	 *
+	 * @param array $args       Positional args.
+	 * @param array $assoc_args Associative args.
+	 */
+	public function cli_command( /** @noinspection PhpUnusedParameterInspection */ $args, $assoc_args ) {
+
+		$is_print     = filter_var( \WP_CLI\Utils\get_flag_value( $assoc_args, 'print', false ), FILTER_SANITIZE_STRING );
+		$is_synthetic = filter_var( \WP_CLI\Utils\get_flag_value( $assoc_args, 'is-synthetic', false ), FILTER_SANITIZE_STRING );
+		$endpoint     = filter_var( \WP_CLI\Utils\get_flag_value( $assoc_args, 'endpoint', '' ), FILTER_SANITIZE_STRING );
+		$endpoint     = untrailingslashit( $endpoint );
+
+		$urls     = filter_var( \WP_CLI\Utils\get_flag_value( $assoc_args, 'urls', false ), FILTER_SANITIZE_STRING );
+		$post_ids = filter_var( \WP_CLI\Utils\get_flag_value( $assoc_args, 'post_ids', false ), FILTER_SANITIZE_STRING );
+		$term_ids = filter_var( \WP_CLI\Utils\get_flag_value( $assoc_args, 'term_ids', false ), FILTER_SANITIZE_STRING );
+
+		$args = [
+			'urls'         => ( ! empty( $urls ) ) ? explode( ',', $urls ) : [],
+			'post_ids'     => ( ! empty( $post_ids ) ) ? explode( ',', $post_ids ) : [],
+			'term_ids'     => ( ! empty( $term_ids ) ) ? explode( ',', $term_ids ) : [],
+			'endpoint'     => $endpoint,
+			'is_synthetic' => $is_synthetic,
+		];
+
+		$support = new SupportData( $args );
+		$data    = $support->get_data();
+
+		if ( $is_print ) {
+
+			// Print the data.
+			$print = strtolower( trim( $is_print ) );
+			if ( 'json' === $print ) {
+				echo wp_json_encode( $data ) . PHP_EOL;
+			} else {
+				echo wp_json_encode( $data, JSON_PRETTY_PRINT ) . PHP_EOL;
+			}
+		} else {
+
+			$response = $support->send_data();
+
+			if ( is_wp_error( $response ) ) {
+				$error_message = $response->get_error_message();
+				\WP_CLI::warning( "Something went wrong: $error_message" );
+			} else {
+				\WP_CLI::success( $response );
```

Comment:
`success` expects a string, but `$response` is an array of `WP_Error`. I'm surprised that PhpStan didn't catch this.